sqlitecluster/SQLiteNode.cpp

```diff
@@ -1414,13 +1412,11 @@ bool SQLiteNode::update(uint64_t& nextActivity) {
                                 _finishCommand(_currentCommand);
                                 _currentCommand = nullptr;
                             }
-
-                            // **NOTE: This loops back and starts the next command of the same priority immediately
-                            if (STimeNow() > nextActivity) {
-                                SINFO("Timeout reached while processing commands. Exceeded by "
-                                      << (STimeNow() - nextActivity) << "us");
-                                return false;
-                            }
+                            // Return false to allow caller to read new commands from the network. This prevents us
+                            // from getting stuck in a state where a long queue of low-priority commands keeps getting
+                            // processed where a queue of higher-priority commands is building up in a network buffer,
+                            // but never gets read and made available to process.
+                            return false;
```

Comment:
Because of the `if/else` statement above, we've either returned `true` already, so that we'll evaluate a processed command for quorum (and then return false after evaluating it), or we'll return false anyway, because we've finished processing a read-only command. Being that we're returning false anyway, the timeout check becomes irrelevant, and has been removed.